playerbot/strategy/values/AttackersValue.cpp

```diff
@@ -116,64 +106,98 @@ void AttackersValue::RemoveNonThreating(set<Unit*>& targets)
 {
+    if(!attacker)
+        return false;
+
     Creature *c = dynamic_cast<Creature*>(attacker);
+    Group* group = player->GetGroup();
+    Player* master = nullptr;
 
     bool rti = false;
-    if (attacker && bot->GetGroup())
-        rti = bot->GetGroup()->GetTargetIcon(7) == attacker->GetObjectGuid();
-
-    PlayerbotAI* ai = bot->GetPlayerbotAI();
-    
     bool leaderHasThreat = false;
-    if (attacker && bot->GetGroup() && ai->GetMaster())
-        leaderHasThreat = attacker->getThreatManager().getThreat(ai->GetMaster());
-
     bool isMemberBotGroup = false;
-    if (bot->GetGroup() && ai->GetMaster() && ai->GetMaster()->GetPlayerbotAI() && !ai->GetMaster()->GetPlayerbotAI()->IsRealPlayer())
-        isMemberBotGroup = true;
+    bool inVehicle = false;
+    bool hasEnemyPlayerTarget = false;
+    bool inDuel = false;
+    bool canSeeAttacker = false;
+    bool hasShackleUndeadAura = false;
+    bool hasAttackTaggedStrategy = false;
+    bool isSapped = false;
+    bool isGouged = false;
+    bool isStunned = false;
+    bool isPolymorphed = false;
+    bool isFeared = false;
+    bool isFriendly = false;
+    bool isDead = false;
+
+    // If the player is a bot
+    PlayerbotAI* bot = player->GetPlayerbotAI();
+    if(bot)
+    {
+        master = bot->GetMaster();
+        if (master)
+        {
+            if(group)
+            {
+                leaderHasThreat = attacker->getThreatManager().getThreat(master);
 
-    bool inCannon = ai->IsInVehicle(false, true);
+                // If the master is a bot
+                if (master->GetPlayerbotAI() && !master->GetPlayerbotAI()->IsRealPlayer())
+                    isMemberBotGroup = true;
+            }
+        }
+
+        inVehicle = bot->IsInVehicle(false, true);
+        hasEnemyPlayerTarget = bot->GetAiObjectContext()->GetValue<Unit*>("enemy player target")->Get();
+        hasAttackTaggedStrategy = bot->HasStrategy("attack tagged", BotState::BOT_STATE_NON_COMBAT);
+        isSapped = bot->HasAura("sap", attacker);
+        isGouged = bot->HasAura("gouge", attacker);
+        hasShackleUndeadAura = bot->HasAura("shackle undead", attacker);
+    }
 
-    bool enemy = ai->GetAiObjectContext()->GetValue<Unit*>("enemy player target")->Get();
-    bool duel = false;
-    if (attacker && bot->duel && bot->duel->opponent && attacker->GetObjectGuid() == bot->duel->opponent->GetObjectGuid())
-        duel = true;
+    isStunned = attacker->IsStunned();
+    isPolymorphed = attacker->IsPolymorphed();
+    isFeared = sServerFacade.IsFeared(attacker);
+    isDead = sServerFacade.UnitIsDead(attacker);
+    isFriendly = sServerFacade.IsFriendlyTo(attacker, player);
+    inDuel = player->duel && player->duel->opponent && (attacker->GetObjectGuid() == player->duel->opponent->GetObjectGuid());
+    canSeeAttacker = attacker->IsVisibleForOrDetect(player, player->GetCamera().GetBody(), true);
+    rti = group && (group->GetTargetIcon(7) == attacker->GetObjectGuid());

```

Comment:
Move the conditions out of the return value for better readability (it is still very hard to read, but it's an improvement...)